### PR TITLE
Incorporate the new CLI layout in builds scripts and add Portable Project Tool commands

### PR DESCRIPTION
**Pull request by Sridhar-MS**
- Layout the Sdk, SharedFx and SharedHost in the stage1 and stage2 dirs.
- Create only 2 zips/tar files 
   - `dotnet` - contains shared fx and shared host.
   - `dotnet-dev` - contains shared fx, cli sdk and shared host.
- There is still work that needs to be done for the installers.
  - Need to create a `dotnet` redist for both osx and windows. This will have sharedfx and shared host.
  - Need to create a `dotnet-dev` installer for osx. This will contains sharedfx, shared host and cli sdk.

cc: @piotrpMSFT @brthor

### CODE REVIEW

**Comment by Sridhar-MS** [comment]
@dotnet-bot test this please

**Comment by anurse** [comment]
It looks like some of the failures are the ones that were fixed by #1884. Has this been rebased recently?

> 14:21:17       System.InvalidOperationException : Cannot start process because a file name has not been provided.
14:21:17     Microsoft.DotNet.Tests.EndToEnd.EndToEndTest.TestDotnetBuild [FAIL]
14:21:17       Stack Trace:
14:21:17            at System.Diagnostics.Process.Start()
14:21:17            at Microsoft.DotNet.Tools.Test.Utilities.TestCommand.RunProcess(String executable, String args, StreamForwarder stdOut, StreamForwarder stdErr)
14:21:17            at Microsoft.DotNet.Tools.Test.Utilities.TestCommand.ExecuteWithCapturedOutput(String args)
14:21:17            at Microsoft.DotNet.Tools.Test.Utilities.TestBase.TestExecutable(String outputDir, String executableName, String expectedOutput)
14:21:17            at Microsoft.DotNet.Tools.Test.Utilities.TestBase.TestOutputExecutable(String outputDir, String executableName, String expectedOutput, Boolean native)
14:21:17            at Microsoft.DotNet.Tests.EndToEnd.EndToEndTest.TestDotnetBuild()

**Comment by Sridhar-MS** [comment]
@anurse This branch has been rebased. I can even see the template changes here in the branch - https://github.com/dotnet/cli/blob/cli-sdk-stages/src/dotnet/commands/dotnet-new/CSharp_Console/project.json.template

**Comment by Sridhar-MS** [comment]
We need to update the End2End tests.
- `dotnet build` on a portable app no longer drops an executable in the output path. 
- `dotnet publish` on a portable app no longer drops an executable in the output path. 

@brthor  is updating the test to reflect this.

**Comment by brthor** [comment]
Made End To End test cases call the muxer rather than the output directly when a runtimeconfig.json is detected. This should get us unblocked for the moment

**Comment by Sridhar-MS** [comment]
@dotnet-bot test this please

**Comment by anurse** [comment]
We need this for ASP.NET porting. Sounds like things are well in hand though :)

File: src/Microsoft.DotNet.ProjectModel/RuntimeConfig/RuntimeConfig.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.IO;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class RuntimeConfig
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/RuntimeConfig/RuntimeConfig.cs]: cc @anurse I think this abstraction will need to move and be less hacky, but for the moment does this appear correct?

**Comment by Sridhar-MS** [comment]
@anurse We are waiting for at least some green in the CI

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    public class ToolPathCalculator
+    {
+        private readonly string _packagesDirectory;
+
+        public ToolPathCalculator(string packagesDirectory)
+        {
+            _packagesDirectory = packagesDirectory;
+        }
+
+        public string GetBestLockFilePath(string packageId, VersionRange versionRange, NuGetFramework framework)
```
Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs]: cc @joelverhagen 

Without the tool baked into the consumer app lockfile yet I needed to put this in place. 
In the long term I'd like to use NuGet's libraries for this.

Comment (joelverhagen) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs]: Understood. I am actually about to send out a PR with tools in the consuming project's lock file.

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by brthor** [comment]
OS X issue fixed in #1937

**Comment by piotrpMSFT** [comment]
latest failure on Ubuntu seems NuGet related. @joelverhagen any ideas?

**Comment by Sridhar-MS** [comment]
The failure on OSX is Nuget related and we are already aware of it.
On ubuntu it is the expected debian package failures.

**Comment by piotrpMSFT** [comment]
cool. Do we know how to address the OS X issue?

btw, debian work is in brthor/fix-deb-path. Want me to push them to this branch so we can all hack?

**Comment by Sridhar-MS** [comment]
@brthor  fixed nuget issue in #1937.

**Comment by Sridhar-MS** [comment]
@piotrpMSFT , this merge is blocking asp.net porting. Should we get this in and then fix the debian packages?

**Comment by piotrpMSFT** [comment]
sure

**Comment by brthor** [comment]
OSX passing locally for Sridhar and I. 

Feeling confident in these changes right now.

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by piotrpMSFT** [comment]
did someone forget to push? It doesn't look like this got rebased atop @brthor's fix yet...

**Comment by brthor** [comment]
branch is rebased, windows, centos were green, only waiting for a green OS X

**Comment by Sridhar-MS** [comment]
OSX build succeeded. Going for the merge..

**Pull request merged**